File: SSServer/src/main/java/com/prose/crhen/SSServer/db/VolumeHistoryRepository.java
package com.prose.crhen.SSServer.db;

import com.prose.crhen.SSServer.model.Volume;
import com.prose.crhen.SSServer.model.VolumeHistory;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface VolumeHistoryRepository extends JpaRepository<VolumeHistory, Long> {

    List<VolumeHistory> findByVolume(Volume volume);
}

File: SSKotlinServer/src/main/kotlin/com/prose/crhen/SSKotlinServer/SSKotlinServer/db/VolumeHistoryRepository.kt
package com.prose.crhen.SSKotlinServer.SSKotlinServer.db

import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.Volume
import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.VolumeHistory
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface VolumeHistoryRepository: JpaRepository<VolumeHistory, Long> {
    fun findByVolume(volume: Volume): List<VolumeHistory>
}
